### DISCUSSION TITLE
Official iframe or Ajax style authentication?

### DISCUSSION THREAD

**Discussion by j-d-carmichael** [keycloak/keycloak#28830] [Q&A]
Is there an official way to authenticate the user with keycloak but without bumping them off to keycloak?

With an app (eg, CapacitorJS) we want the user to stay within the mobile app to do anything with auth, but KeyCloak out of the box does not seem to have an official way to allow this to happen.

**Comment by jonkoops** [comment]
https://www.keycloak.org/docs/latest/securing_apps/#using-the-adapter

**Reply by j-d-carmichael** [reply]
We decided to use zitadel.

The docs here are making it too time-consuming to work with. And if the community response is to a page in the docs that doesn't really answer the question.. it isn't 2010 in the forums anymore.

**Reply by jonkoops** [reply]
From the docs:

> You can configure a silent check-sso option. With this feature enabled, your browser will not perform a full redirect to the Keycloak server and back to your application, but this action will be performed in a hidden iframe. Therefore, your application resources are only loaded and parsed once by the browser, namely when the application is initialized and not again after the redirect back from Keycloak to your application. This approach is particularly useful in case of SPAs (Single Page Applications).

If you need to do this in a native app outside of HTML then you'll have to use whatever client for that platform is compatible with OpenID Connect.
